lib/senkyoshi/models/content_file.rb

```diff
@@ -26,10 +26,15 @@ def self.correct_linkname(canvas_file)
       canvas_file.file_path.split("/").last
     end
 
-    def canvas_conversion(canvas_file = nil)
-      path = canvas_file ? canvas_file.file_path : @linkname
+    def canvas_conversion(resources, canvas_file = nil)
+      path = if canvas_file
+               canvas_file.file_path
+             else
+               resource = resources.detect_xid(@name)
+               resource.path if resource
```

Comment:
Since it is only available in ruby 2.3 and we need to support ruby 2.0, 2.1, 2.2, and 2.3